### DISCUSSION TITLE
import @chainlink won't work even after toml

### DISCUSSION THREAD

**Discussion by DukeLurkin** [Cyfrin/foundry-full-course-cu#448] [Q&A]
Pictures attached.  The files cannot be found.  This is my second time going through this part of the course.  I'm on a laptop.  Didn't have this issue on my desktop last week.  Forge compile keeps telling me nothing has been changed even if I completely delete a file, that doesn't seem right. 
![toml](https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/cee86ebd-f834-4868-8bed-082b26c42981)
![priceConverter](https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/b2826d66-023d-4e32-a52e-0df8722e3d38)
![fundMe](https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/e69022ba-456f-4923-811a-598abc42142b)
![forgeCompile](https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/68e87cbd-8f0f-47aa-9afc-d323e7f8f82d)

**Comment by DukeLurkin** [comment]
Also, if I hard code it, then it changes and cannot find priceConverter.sol when it could moments before lol
![fundMehardcode](https://github.com/Cyfrin/foundry-full-course-f23/assets/123668611/30725507-b235-4720-bfce-4ba78a0e9185)

**Comment by ChefAharoni** [comment]
Hey, try the following solutions:
https://github.com/Cyfrin/foundry-full-course-f23/discussions/271#discussioncomment-6378487

https://github.com/Cyfrin/foundry-full-course-f23/discussions/194#discussioncomment-6315633

**Reply by DukeLurkin** [reply]
Ty for this.  That's almost 100% the same problem.  Once I get back on my laptop I'll try that out.

**Comment by theirrationalone** [comment]
Check "lib" folder after installing packages using NPM(Node Package Manager). If "NPM"pulled packages successfully then they might be in the "lib" folder. And if you found nothing from your installed packages into "lib" folder then check Your project directory structure and if you found a folder named "node_modules" then this means there is no support for foundry from "NPM" community.

You can do two things in this condition....

Solution 1:
- Go to "node_modules/chainlink/contracts" folder followed by the "node_modules" folder,
- Copy or cut everything inside the contracts folder,
- Now comeback to the top level to your project's directory and go to the "lib" folder,
- Create a folder named as "chainlink" then open it and again create another folder named as "contracts",
- Open "Contracts" folder and Paste everything you copied into lib folder (inside the lib folder).
- And voila, You're good to go.
- Now You just have to re-set mappings i.e.,

## **In foundry.toml file**

```code
remappings = ["@chainlink/contracts/=lib/chainlink/contracts/"].
``` 

Solution 2:
Just delete that "node_modules" folder and execute this command below on terminal.

```terminal
forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit
```
